### PR TITLE
new fcast year approach

### PR DESCRIPTION
**Pull request by Rick-Methot-NOAA**
## Concisely describe what has been changed/addressed in the pull request.

* Resolves issue #493 

## What tests have been done? 
passing the GHA tests.
also did character-level compare of report, echoinput and forecast-report to identify any differences.  All are related to intended changes in text; no meaningful diffs in numeric values.  This test is what facilitated the fix for vermillion.
## Is there an input change for users to Stock Synthesis? 

 [x] The input change is in the related issue(s). -->

### CODE REVIEW

**Comment by e-gugliotti-NOAA** [comment]
@Rick-Methot-NOAA - just made some changes to the workflows in PR #496 so that workflows won't be triggered if it is a draft PR. If you look at the bottom of the PR page, the only workflow triggered by the changes made was skipped because it is now a draft PR.

**Comment by Rick-Methot-NOAA** [comment]
First test is nearly all passing.  Only the forecast catch for timevary growth and vermillion snapper are showing a small delta that cause failure.  I'll do a deep dive on that next.

**Comment by iantaylor-NOAA** [comment]
I'm happy to review this but won't be able to get it done until after Sept 4 (I'm in FIMS meeting this week and have a manuscript review due, then going on leave Friday and next week).

**Comment by Rick-Methot-NOAA** [comment]
Let's see if Brian can help out if Chantel cannot get to this either.
@chantelwetzel-noaa

File: SS_write_ssnew.tpl [hunk 1]
```diff
@@ -1537,10 +1537,40 @@ FUNCTION void write_nucontrol()
   NuFore << "# where none and simple require no input after this line; simple sets forecast F same as end year F" << endl;
   NuFore << N_Fcast_Yrs << " # N forecast years " << endl;
   NuFore << Fcast_Flevel << " # Fmult (only used for Do_Forecast==5) such that apical_F(f)=Fmult*relF(f)" << endl;
-  NuFore << "#_Fcast_years:  beg_selex, end_selex, beg_relF, end_relF, beg_mean recruits, end_recruits  (enter actual year, or values of 0 or -integer to be rel. endyr)" << endl
+  anystring = "";
+  if(Fcast_yr_rd(1) != -12345)
+  {   //  write in old format
+    NuFore << "#_Fcast_years for averaging:  beg_selex, end_selex, beg_relF, end_relF, beg_mean recruits, end_recruits  (enter actual year, or values of 0 or -integer to be rel. endyr)" << endl
```
Comment (chantelwetzel-noaa) [inline SS_write_ssnew.tpl:1543]: In other places in the revised code, we use the word "mean" but here we use "average". I would suggest using one or the other for consistency.

File: SS_write_ssnew.tpl [hunk 2]
```diff
@@ -1556,75 +1586,29 @@ FUNCTION void write_nucontrol()
     }
   }
 
-  NuFore << Fcast_Loop_Control(1) << " #_N forecast loops (1=OFL only; 2=ABC; 3=get F from forecast ABC catch with allocations applied)" << endl;
-  NuFore << Fcast_Loop_Control(2) << " #_First forecast loop with stochastic recruitment" << endl;
-  NuFore << Fcast_Loop_Control(3) << " #_Forecast recruitment:  0= spawn_recr; 1=mult*spawn_recr_fxn; 2=mult*VirginRecr; 3=recent mean for recr and recr_dist from yr range above; 4=mult * recent mean for recr only" << endl;
-  NuFore << "#_ for options 3 and 4, need to set phase for fore_recr_devs to -1 in control to get requested mean in MCMC" << endl;
+  NuFore << "#" << endl << Fcast_Loop_Control(1) << " #_N forecast loops (1=OFL only; 2=ABC; 3=get F from forecast ABC catch with allocations applied)" << endl;
+  NuFore << Fcast_Loop_Control(2) << " # First forecast loop with stochastic recruitment" << endl;
+  NuFore << Fcast_Loop_Control(3) << " # Forecast base recruitment:  0= spawn_recr; 1=mult*spawn_recr_fxn; 2=mult*VirginRecr; 3=deprecated; 4=mult*average_over_yr_range" << endl;
```
Comment (chantelwetzel-noaa) [inline SS_write_ssnew.tpl:1591]: "average_over_yr_range" revise to "mean_over_yr_range"

File: SS_write_ssnew.tpl [hunk 3]
```diff
@@ -1556,75 +1586,29 @@ FUNCTION void write_nucontrol()
...
+  NuFore << "# for option 4, set phase for fore_recr_devs to -1 in control to get constant mean in MCMC, else devs will be applied" << endl;
```
Comment (chantelwetzel-noaa) [inline SS_write_ssnew.tpl:1592]: I would revise for clarity "for forecast base recruitment option 4 (mult*mean_over_yr_range),"

**Review by chantelwetzel-noaa — CHANGES_REQUESTED** [review]
I have only suggested minor revisions here for clarity but it also appears that the model run actions are failing.  Is this still a draft pull request?  I looked around but did not see "draft" anywhere and based on the comments it looks like @e-gugliotti-NOAA fixed the actions for draft requests. Can someone confirm whether we would expect the model run actions to be passing or failing here?

**Comment by Rick-Methot-NOAA** [comment]
The failing runs are all due to time-varying growth and in vermillion snapper.  I think this is not due to these new changes and I'll investigate.

**Comment by Rick-Methot-NOAA** [comment]
something about the new code's handling of vermillion test case is still off.  Working on it.

**Review by chantelwetzel-noaa — APPROVED** [review]
These additional revisions look good to me and it is great we are now successfully passing all the checks.

**Pull request merged**